tools/pkg/repo/publish.py

```diff
@@ -0,0 +1,653 @@
+"""
+These commands are used to build the pacakge repository files.
+"""
+# pylint: disable=resource-leakage,broad-except,3rd-party-module-not-gated
+from __future__ import annotations
+
+import fnmatch
+import json
+import logging
+import os
+import pathlib
+import re
+import sys
+import tempfile
+import textwrap
+from typing import TYPE_CHECKING, Any
+
+import packaging.version
+from ptscripts import Context, command_group
+
+import tools.pkg
+import tools.utils
+from tools.utils import (
+    Version,
+    create_full_repo_path,
+    get_repo_json_file_contents,
+    get_salt_releases,
+    parse_versions,
+)
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+except ImportError:
+    print(
+        "\nPlease run 'python -m pip install -r "
+        "requirements/static/ci/py{}.{}/tools.txt'\n".format(*sys.version_info),
+        file=sys.stderr,
+        flush=True,
+    )
+    raise
+
+log = logging.getLogger(__name__)
+
+publish = command_group(
+    name="publish",
+    help="Packaging Repository Publication Related Commands",
+    parent=["pkg", "repo"],
+)
+
+
+@publish.command(
+    arguments={
+        "repo_path": {
+            "help": "Local path for the repository that shall be published.",
+        },
+        "salt_version": {
+            "help": "The salt version for which to build the repository",
```

Comment:
Is it building the repository here? Or publishing? This looks like a copy/paste from the `create.py`